### ISSUE TITLE
Offloading/Rollback Improvements

### ISSUE THREAD

**Issue by HardeepAsrani** [Codeinwp/optimole-wp#665]
### What problem does this address?

Right now the process of offloading/rollback is opaque, and it makes it harder to debug issues for the customers. Even if we want to run big tests, it often requires you to monitor how the process is going, and it's not ideal.

### What is your proposed solution?

Here are a few improvements that can help us improve the process:

### 1. Adding Confirmation dialog

As Andrie mentioned, adding a confirmation modal will be a great step to make sure users don't by mistake, start/stop the progress while trying to explore the dashboard. It doesn't require much effort from our side, either.

### 2. Progress Logging

Similar to how security/backup plugins do, we can create a log of the ongoing progress so the user, as well as us, when trying to debug, can observe how the progress is going and if any errors are generated at any point. This should help us debug when things go wrong or observe when the progress takes longer than expected, including having a start/end time with the number of images covered to give us a better idea.

An example of something similar is from the Wordfence plugin, which is shared below as a screenshot:
<img width="1195" alt="Screenshot 2023-09-13 at 11 48 38 AM" src="https://github.com/Codeinwp/optimole-wp/assets/2649903/7b500da4-302d-4068-a216-7c349d132fb3">

Of course, the logs should be separate for both offloading/rollback so we can observe them better.

One thing that we need to research is how these plugins save their logs. Is it saved on the database or as a log file? Should the logs be publicly sharable? If yes then do we use the image URL or use image_id to avoid exposing image URLs, in case the website is private?

### Will this feature require documentation? (Optional)

Yes, it requires documentation.

**Comment by abaicus** [comment]
@HardeepAsrani, I will close #655 in place of this one. Adding more information and an estimate to this confirmation dialogue itself would be of further benefit.

We can probably do a public log in the `uploads` folder (similar to how we do it on the Demo Data Exporter plugin) for easy access.

Furthermore, we can improve our request responses when polling the offload/rollback progress. Right now, we only return the `count`, which is not very explicit.

**Comment by HardeepAsrani** [comment]
@abaicus Estimation is pretty hard to get, especially in the current workflow where we don't just upload images but also fetch posts to scan for images, so it fluctuates based on the total number of posts & number of posts remaining. Even before the new offloading flow, the estimation was wildly inaccurate (off by hours during large tests).

When we implement the option to upload images, then we might get a better estimation based on the number of total images, but with the current flow, it's just a wild guess that might or might not be close.

**Comment by abaicus** [comment]
Yes, in the context where we move to the new offloading process. Then, we can estimate based on the image count rather than taking all the things into consideration.

**Comment by GrigoreMihai** [comment]
I also wanted to add a suggestion about this, if you think it's worth implementing. 

For debug  we also have an endpoint in AWS were we can log some messages for easier access.

Currently we are only logging the errors when the optimized image request is failing during the offload, for example: https://vertis.d.pr/i/zbIGQp . 

We are also having various logs inside the offload code in the places were it could fail, but those were pretty hard to access until now since were logged on the user site. 

My suggestion would have been that when we enable those logs to also send them to this endpoint, with some specific tags maybe for easier querying.
